simtools/camera_efficiency.py

```diff
@@ -0,0 +1,383 @@
+                    float(words[0])
+                except:
+                    continue
+
+                if float(words[0]) < 200 or float(words[0]) > 1000:
+                    continue
+                numbers = [float(w) for w in words]
+                for i in range(len(effPars) - 10):
+                    self._results[effPars[i]].append(numbers[i])
+                C1 = numbers[8] * (400 / numbers[0])**2
+                C2 = C1 * numbers[4] * numbers[5]
+                C3 = C2 * numbers[6] * numbers[7]
+                C4 = C3 * numbers[3]
+                C4x = C1 * numbers[3] * numbers[6] * numbers[7]
+                self._results['C1'].append(C1)
+                self._results['C2'].append(C2)
+                self._results['C3'].append(C3)
+                self._results['C4'].append(C4)
+                self._results['C4x'].append(C4x)
+                N1 = numbers[14]
+                N2 = N1 * numbers[4] * numbers[5]
+                N3 = N2 * numbers[6] * numbers[7]
+                N4 = N3 * numbers[3]
+                N4x = N1 * numbers[3] * numbers[6] * numbers[7]
+                self._results['N1'].append(N1)
+                self._results['N2'].append(N2)
+                self._results['N3'].append(N3)
+                self._results['N4'].append(N4)
+                self._results['N4x'].append(N4x)
+
+        self._hasResults = True
+        if export:
+            self.exportResults()
+    # END of analyze
+
+    def exportResults(self):
+        ''' Export results to a csv file. '''
+        if not self._hasResults:
+            self._logger.error('Cannot export results because it does not exist')
+        else:
+            self._logger.info('Exporting results to {}'.format(self._fileResults))
+            table = Table(self._results)
+            ascii.write(table, self._fileResults, format='basic', overwrite=True)
+
+    def _readResults(self):
+        ''' Read existing results file and store it in _results. '''
+        table = ascii.read(self._fileResults, format='basic')
+        self._results = dict(table)
+        self._hasResults = True
+
+    def plot(self, key, **kwargs):
+        '''
+        Plot key vs wavelength.
+
+        Parameters
+        ----------
+        key: str
+            cherenkov or nsb
+        **kwargs:
+            kwargs for plt.plot
+
+        Raises
+        ------
+        KeyError
+            If key is not among the valid options.
+        '''
+        if key not in ['cherenkov', 'nsb']:
+            msg = 'Invalid key to plot'
+            self._logger.error(msg)
+            raise KeyError(msg)
+
+        ax = plt.gca()
+        firstLetter = 'C' if key == 'cherenkov' else 'N'
+        for par in ['1', '2', '3', '4', '4x']:
+            ax.plot(
+                self._results['wl'],
+                self._results[firstLetter + par],
+                label=firstLetter + par,
+                **kwargs
+            )
+
+    def plotCherenkovEfficiency(self):
```

Comment:
Why the duplication of "plot" and the two specific functions? That is, why not call the specific functions from "plot"? 
In the meantime I will change the specific functions and once we resolve this I might change "plot" as well.

Reply:
This is similar to what was discussed before concerning these plot functions.
I think it's good to have the simple "plot" function in which we just pipe the data into the plt plot functions and leave all the rest to the user. They are useful for testing, at least. I would keep them by now, no harm.